instrumentation/opentelemetry-instrumentation-fastapi/tests/test_fastapi_instrumentation.py

```diff
@@ -170,7 +170,8 @@ def setUp(self):
         self._instrumentor = otel_fastapi.FastAPIInstrumentor()
         self._app = self._create_app()
         self._app.add_middleware(HTTPSRedirectMiddleware)
-        self._client = TestClient(self._app)
+        self._client = TestClient(self._app, base_url="https://testserver:443")
```

Comment:
This was the root cause of tests failures. It turns out _every request was being made twice because it was getting redirected to https_. Before this PR that wasn't being instrumented correctly, so this was not being caught! I think that's another major bug this PR is fixing.